### PR TITLE
Optionally request verification before sending messages

### PR DESCRIPTION
**Pull request by jonmattgray**
User can optionally supply a `requestProof` of type `CreateProofRequestOptions` when they `POST /basic-messages` This means they can ask for proof for whatever they like about the agent they're sending the message to. The message only sends if the proof verification passes.

Can also set the timeout length to wait for proof verification. Defaults to 1000ms

### CODE REVIEW

File: src/controllers/basic-messages/BasicMessageController.ts [hunk 1]
```diff
@@ -48,4 +66,25 @@ export class BasicMessageController extends Controller {
       throw error
     }
   }
+
+  async waitForProof(connectionId: string, proofId: string, timeoutMs: number): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        reject(new HttpResponse({ message: `Proof request to connection '${connectionId}' timed out` }))
```
Comment (msterle) [inline src/controllers/basic-messages/BasicMessageController.ts:73]: It's good practice to remove the `onProofStateChanged` listener here as well. Although a promise can only resolve once, the listener will remain attached.

File: tests/unit/basicMessage.test.ts [hunk 2]
```diff
@@ -44,6 +48,83 @@ describe('BasicMessageController', () => {
...
+      expect(response.statusCode).to.be.equal(500)
+      expect(response.body).to.contain('timed out')
+    })
+
+    test('should give 400 when attempt to message an unverified agent ', async () => {
+      const testProof = getTestProof()
+      const requestProofStub = stub(bobAgent.proofs, 'requestProof')
+      requestProofStub.resolves(testProof)
+
+      setTimeout(
```
Comment (msterle) [inline tests/unit/basicMessage.test.ts:102]: 👍 this use of timeout is absolutely fine

**Review by msterle — CHANGES_REQUESTED** [review]
minor change, but looking good otherwise

File: src/controllers/basic-messages/BasicMessageController.ts (hunk 1 above)
Comment (jonmattgray) [reply src/controllers/basic-messages/BasicMessageController.ts:74]: added the removal

**Pull request closed without merging**